File: android-agent/src/main/java/io/opentelemetry/android/instrumentation/RumScreenName.java
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.android.instrumentation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * This annotation can be used to customize the {@code screen.name} attribute for an instrumented
 * Fragment or Activity.
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface RumScreenName {
    /** Return the customized screen name. */
    String value();
}

File: instrumentation/common-api/src/main/java/io/opentelemetry/android/instrumentation/annotations/RumScreenName.kt
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.android.instrumentation.annotations

/**
 * This annotation can be used to customize the `screen.name` attribute for an instrumented
 * Fragment or Activity.
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class RumScreenName(
    /** Return the customized screen name.  */
    val value: String,
)
